examples/k-means/k-means.hpp

```diff
@@ -172,22 +173,22 @@ class KMeansModel
 //! input and an output parameter. The method returns a std::pair<Point2D,
 //! size_t> = Point2DClusterId into the centroids for each input point.
 template <typename Point, typename InStack>
-auto KMeans(const DIA<Point, InStack>& input_points,
-            size_t dimensions, size_t num_clusters, size_t iterations) {
+auto KMeans(const DIA<Point, InStack>& input_points, size_t dimensions,
+            size_t num_clusters, size_t iterations, double epsilon) {
 
     auto points = input_points.Cache();
 
+	bool break_condition = false;
+
     using ClosestCentroid = ClosestCentroid<Point>;
     using CentroidAccumulated = CentroidAccumulated<Point>;
 
     DIA<Point> centroids = points.Keep().Sample(num_clusters);
 
-    for (size_t iter = 0; iter < iterations; ++iter) {
+	std::vector<Point> local_centroids = centroids.AllGather();
```

Comment:
This looks overly complex... you need only one variable outside of the loop. Because you get new centroids in the loop in line 232.